### PR TITLE
port toranj test39 address_cache_table_snoop

### PR DESCRIPTION
**Pull request by priyankaCh22**
Please find attached PASS logs.
[TestAddressCacheTableSnoop.log](https://github.com/openthread/silk/files/5505666/TestAddressCacheTableSnoop.log)

### CODE REVIEW

File: silk/node/wpantund_base.py [hunk 1]
```diff
@@ -476,20 +478,22 @@ def timed_ping6(self, ipv6_target, num_pings, payload_size=8, interface=None):
...
+    def send_udp_data(self, target: str, port: int, message: str, source: str = None,
+                      src_port: int = random.randint(11200, 11400), hop_limit: int = 64, timeout: float = 10):
         """Perform netcat command to send UDP message to ipv6_target via port.
 
         Args:
             target (str): target address.
             port (int): target port.
             message (str): message to send.
             source (str, optional): source address. Defaults to None.
+            src_port (int, optional): source port. Defaults to None.
```
Comment (qilinkang) [inline silk/node/wpantund_base.py:506]: default to a random port between 11200 and 11400 ?

File: silk/node/wpantund_base.py [hunk 2]
```diff
@@ -128,7 +129,7 @@ def un_allowlist_node(self, item):
     #   wpan_node functionality
     #################################
 
-    def form(self, network, role, xpanid=None, panid=None, mesh_local_prefix=None):
+    def form(self, network, role, xpanid=None, panid=None, mesh_local_prefix=None, test=None):
         """
```
Comment (qilinkang) [inline silk/node/wpantund_base.py:135]: please add Args doc string.

File: silk/node/wpantund_base.py [hunk 3]
```diff
@@ -502,3 +506,16 @@ def receive_udp_data(self, port: int, message: str, timeout: int = 10):
         command = f"nc -6lu {port}"
 
         self.make_netns_call_async(command, message, timeout=timeout, exact_match=True)
+
+    def receive_no_udp_data(self, port: int, message: str, timeout: int = 10):
```
Comment (qilinkang) [inline silk/node/wpantund_base.py]: can use receive_udp_data() with message="" ?

File: silk/node/wpantund_base.py (hunk 1 above)
Comment (priyankaCh22) [reply silk/node/wpantund_base.py:506]: Ack

File: silk/node/wpantund_base.py (hunk 2 above)
Comment (priyankaCh22) [reply silk/node/wpantund_base.py:135]: Ack

File: silk/node/wpantund_base.py (hunk 3 above)
Comment (priyankaCh22) [reply silk/node/wpantund_base.py]: Ack

File: silk/node/wpantund_base.py [hunk 4]
```diff
@@ -128,9 +128,17 @@ def un_allowlist_node(self, item):
     #   wpan_node functionality
     #################################
 
-    def form(self, network, role, xpanid=None, panid=None, mesh_local_prefix=None):
+    def form(self, network, role, xpanid=None, panid=None, mesh_local_prefix=None, add_ip_addr="Yes"):
         """
         Form the PAN specified by the arguments.
+        Args:
```
Comment (qilinkang) [inline silk/node/wpantund_base.py:137]: Please use this format for consistency.

Args:
            target (str): target address.

File: silk/node/wpantund_base.py [hunk 5]
```diff
@@ -157,12 +165,18 @@ def form(self, network, role, xpanid=None, panid=None, mesh_local_prefix=None):
 
         self.__get_network_properties("form", network)
 
-        self._get_addr("form")
+        if add_ip_addr == "Yes":
+            self._get_addr("form")
 
-    def join(self, network, role):
+    def join(self, network, role, add_ip_addr="Yes"):
```
Comment (qilinkang) [inline silk/node/wpantund_base.py]: suggest use a boolean: add_ip_addr=True

File: silk/node/wpantund_base.py (hunk 4 above)
Comment (priyankaCh22) [reply silk/node/wpantund_base.py:137]: Ack. Done

File: silk/node/wpantund_base.py (hunk 2 above)
Comment (qilinkang) [reply silk/node/wpantund_base.py:135]: Should be "add_ip_addr=True" ?

File: silk/node/wpantund_base.py [hunk 6]
```diff
@@ -160,12 +168,18 @@ def form(self, network, role, xpanid=None, panid=None, mesh_local_prefix=None):
 
         self.__get_network_properties("form", network)
 
-        self._get_addr("form")
+        if add_ip_addr == "True":
```
Comment (qilinkang) [inline silk/node/wpantund_base.py]: if add_ip_addr:

File: silk/node/wpantund_base.py [hunk 7]
```diff
@@ -160,12 +168,18 @@ def form(self, network, role, xpanid=None, panid=None, mesh_local_prefix=None):
...
+            self._get_addr("form")
 
-    def join(self, network, role):
+    def join(self, network, role, add_ip_addr="True"):
```
Comment (qilinkang) [inline silk/node/wpantund_base.py]: add_ip_addr=True

File: silk/node/wpantund_base.py [hunk 8]
```diff
@@ -186,7 +200,8 @@ def join(self, network, role):
 
         self.__get_network_properties("join", network)
 
-        self._get_addr("join")
+        if add_ip_addr == "True":
```
Comment (qilinkang) [inline silk/node/wpantund_base.py]: if add_ip_addr:

File: silk/node/wpantund_base.py [hunk 9]
```diff
@@ -493,7 +508,7 @@ def send_udp_data(self, target: str, port: int, message: str, source: str = None
         """
```
Comment (qilinkang) [inline silk/node/wpantund_base.py:509]: missing Doc string for timeout ?

File: silk/node/wpantund_base.py (hunk 7 above)
Comment (priyankaCh22) [reply silk/node/wpantund_base.py]: Ack

File: silk/node/wpantund_base.py (hunk 8 above)
Comment (priyankaCh22) [reply silk/node/wpantund_base.py]: Ack

File: silk/node/wpantund_base.py (hunk 9 above)
Comment (priyankaCh22) [reply silk/node/wpantund_base.py:509]: Ack

File: silk/node/wpantund_base.py (hunk 6 above)
Comment (priyankaCh22) [reply silk/node/wpantund_base.py]: Ack

**Review by qilinkang — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]

**Pull request merged**